### DISCUSSION TITLE
Share an email thread to workspace members

### DISCUSSION THREAD

**Discussion by Bonapara** [twentyhq/twenty#7639] [Ideas]
# 🚨 Design is Outdated

As the `close, expand, minimize` buttons will move to the right edge of the navbar, we need to find another place for the email sharing menu.

# Product behavior

### 1- I click on the people chip on a thread panel

There are 3 different states for this chip:

The thread is shared with `some` other workspace member:
![CleanShot 2024-02-27 at 08 55 58](https://github.com/twentyhq/twenty/assets/19412894/d3668194-461e-4d43-8a4e-c93e602d842e)

This thread has not been shared `with anyone` (I'm the only one with access to this thread):
![CleanShot 2024-02-27 at 09 06 00](https://github.com/twentyhq/twenty/assets/19412894/a8d74673-0f34-4707-8df7-cd9e1c168221)

This thread has been shared `with every workspace member`:
![CleanShot 2024-02-27 at 09 07 52](https://github.com/twentyhq/twenty/assets/19412894/222daab5-985b-4738-bc06-0f12ae94f737)

### 2- A menu shows all workspace members with full email access.

I can click on a people menu item to revoke access from a teammate. (You don't have to specifically click on "-"

![CleanShot 2024-02-27 at 08 57 13](https://github.com/twentyhq/twenty/assets/19412894/65b2001e-001f-46fc-a0ca-ccf8cc11afb5)

### 3- Click "Add participant" to share access with individuals or everyone. 

Same behavior as 2- but clicking on a workspace member menu item gives access to the thread. The person is added to the list in 2-

![CleanShot 2024-02-27 at 09 00 19](https://github.com/twentyhq/twenty/assets/19412894/f107ae64-790b-4045-8431-742241f10c28)

# Figma

![CleanShot 2024-02-27 at 09 12 00](https://github.com/twentyhq/twenty/assets/19412894/5e4cbbc7-a676-42e6-9261-eec23142a169)

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=16093-65962&mode=design&t=MAo8fwzMuDFxsQqD-11

**Comment by FelixMalfait** [comment]
Table structure

**MessageThreadMembers**
- id, createdAt, updatedAt
- messageThreadId - string fk
- workspaceMemberId - fk to workspaceMembers

We'll also need to add a column on MessageThread for the "Everyone" feature

**Comment by pereira0x** [comment]
Can I and @simaosanguinho be assigned to this?

**Comment by FelixMalfait** [comment]
Sure! Thanks. Keep us updated.

**Comment by simaosanguinho** [comment]
I am working with @pereira0x on this issue! Could you also assign me to it?
